app/scripts/patient-dashboard/directives/patient-relationship.directive.js

```diff
@@ -0,0 +1,228 @@
+(function(){
+'use strict';
+angular
+      .module('app.patientdashboard')
+      .directive('patientRelationship',patientRelationship);
+function patientRelationship(){
+  var patientRelationshipDefinition = {
+      restrict: 'EA',
+      templateUrl: 'views/patient-dashboard/patient-relationships-template.html',
+      controller: PatientRelationshipCtrl
+  };
+
+  return patientRelationshipDefinition;
+}
+PatientRelationshipCtrl.$inject = ['$rootScope', '$scope', '$stateParams', 'OpenmrsRestService','$state','dialogs', '$timeout'];
+function PatientRelationshipCtrl($rootScope, $scope, $stateParams, OpenmrsRestService,$state,dialogs, $timeout){
+  $scope.patientRelationships = undefined;
+  $scope.showSuccessAlert = false;
+  $scope.successAlert='';
+  var patientRelationship = getPatientRelationships();
+  var patientRelationshipType=OpenmrsRestService.getPatientRelationshipTypeService()
+  .getPatientRelationshipTypes(function(data){
+    $scope.patientRelationshipTypes=data;
+  },function(error){
+    console.log("The request failed because of ",error);
+  });
+  $scope.loadPatient = loadPatient;
+  function loadPatient(patientUuid) {
+    /*
+      Get the selected patient and save the details in the root scope
+      so that we don't do another round trip to get the patient details
+      */
+      console.log("patient uuid clicked id ",patientUuid);
+    OpenmrsRestService.getPatientService().getPatientByUuid({ uuid: patientUuid },
+           function(data) {
+             $rootScope.broadcastPatient = data;
+             $state.go('patient', { uuid: patientUuid });
+           }
+      );
+  }
+  $scope.setPatientRelationship=setPatientRelationship;
+  function setPatientRelationship(personA,relationshipTypeUUid,personB){
+    if(angular.isDefined(relationshipTypeUUid) && $scope.patient.uuid()!==personB){
+    var relationshipExists=false;
+    _.find($scope.patientRelationships.relationships,function(relationship){
+      console.log("relationship in find loop is ",relationship.uuId());
+      if(personB==relationship.personUuId()) {
+        relationshipExists=true;
+        var patientRelationshipPayload={
```

Comment:
For readability, you want to take this to a function whose purpose is to create payload for relationships.